### PR TITLE
Convert to github actions

### PR DESCRIPTION
**Pull request by rluvaton**
As discussed (https://github.com/goldbergyoni/nodebestpractices/issues/896#issuecomment-826141118), I've migrated from Travis.CI to GitHub Actions

I've sinned, I've also Fixed that the tests would run in this repo as well and not only in the forked ones (https://github.com/goldbergyoni/nodebestpractices/issues/896#issuecomment-810954302)

- [x] Remove `.travis.yml` file

### CODE REVIEW

File: .github/workflows/gen-html.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      uses: actions/setup-node@v2.1.2
+      with:
+        node-version: '14'
+
+    - run: cd .operations && npm install
+    - run: cd .operations && npm run lint
+    - run: cd .operations && npm run build
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        IS_FORK: ${{ github.event.pull_request.head.repo.full_name != 'goldbergyoni/nodebestpractices' }}
```
Comment (rluvaton) [inline .github/workflows/gen-html.yml:29]: haven't checked but this may cause problems if we run on a push event

Comment (rluvaton) [reply .github/workflows/gen-html.yml]: Fixed

**Review by elektronik2k5 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v2.1.2
+      with:
+        node-version: '14'
+
+    - run: cd .operations && npm install
```
Comment (goldbergyoni) [inline .github/workflows/ci.yml:24]: Consider:

```
defaults:
  run:
    shell: bash
    working-directory: .operations
    ```

File: .operations/gen-html.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ const { readdir, readFile, writeFile } = require('graceful-fs');
 const imagemin = require('imagemin');
```
Comment (goldbergyoni) [inline .operations/gen-html.js:7]: @rluvaton Can you kindly remind me what is this file about? Is this new one or an existing file?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - run: cd .operations && npm run lint
```
Comment (goldbergyoni) [inline .github/workflows/ci.yml:25]: Should there be a linting issue, will we see it within the code in the GitHub "files" tab?

**Review by goldbergyoni — CHANGES_REQUESTED** [review]
@rluvaton Great stuff here. Don't have any major concern, mostly asked few questions to refresh my memory

File: .github/workflows/ci.yml (hunk 4 above)
Comment (rluvaton) [reply .github/workflows/ci.yml:25]: I don't think so, but I can check now by introducing a linting issue

File: .operations/gen-html.js (hunk 3 above)
Comment (rluvaton) [reply .operations/gen-html.js:7]: This is an existing file and I think it's about generating GitHub Pages, but I'm not sure 🤔

Comment (goldbergyoni) [reply .operations/gen-html.js:7]: @BrunoScheufler @js-kyle Do you remember what is the purpose of this file? Is it needed anymore?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (goldbergyoni) [reply .github/workflows/ci.yml]: It's possible, should work by default given the default eslint behaviour

This extension enriches the DX:

https://github.com/marketplace/actions/lint-action

Comment (rluvaton) [reply .github/workflows/ci.yml]: When we add eslint we should the GitHub action you suggested, but right now we're using `markdownlint` which doesn't support it (I haven't found a GitHub Action that support that, but I can keep looking)

**Review by goldbergyoni — APPROVED** [review]
LGTM

File: .operations/gen-html.js (hunk 3 above)
Comment (rluvaton) [reply .operations/gen-html.js:7]: Do you want to wait for them to answer or can we merge now?

**Comment by goldbergyoni** [comment]
@rluvaton Feel free to merge

**Pull request merged**